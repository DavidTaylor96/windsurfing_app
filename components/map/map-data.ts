export interface IMarker {
  id: string;
  latitude: number;
  longitude: number;
  title: string;
  description: string;
}


export const markers: IMarker[] = [
  {
    id: '123',
    latitude: 55.953251,
    longitude: -3.188267,
    title: 'one',
    description: 'something new',
  },
  {
    id: '124',
    latitude: 55.853251,
    longitude: -2.488267,
    title: 'two',
    description: 'something new',
  },
  {
    id: '125',
    latitude: 55.753251,
    longitude: -2.588267,
    title: 'Three',
    description: 'something new',
  },
];
