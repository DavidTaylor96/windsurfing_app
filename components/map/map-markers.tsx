import { Fontisto } from '@expo/vector-icons';
import React, { FC } from 'react';
import { Marker } from 'react-native-maps';
import { useMarkerPressed } from '../../hooks/marker-pressed';
import { useMapNavigation } from '../../hooks/use-map-navigation';
import { markers } from './map-data';

interface IMapMarkersProps {}

export const MapMarkers: FC<IMapMarkersProps> = ({}) => {

  const { pressed, setPressed } = useMarkerPressed();

  const {animateToCoordinat } = useMapNavigation();
  return (
    <>
      {markers.map((marker) => (
        <Marker
          onPress={() => {
            animateToCoordinat(marker.latitude, marker.longitude);
            setPressed(!pressed);
          }}
          key={marker.id}
          identifier={marker.id}
          coordinate={{
            latitude: marker.latitude,
            longitude: marker.longitude,
          }}
          title={marker.title}
        >
          <Fontisto name="map-marker-alt" size={29} color="#bf0f2b" />
        </Marker>
      ))}
    </>
  );
};
