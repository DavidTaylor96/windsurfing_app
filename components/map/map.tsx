import { Fontisto } from '@expo/vector-icons';
import * as React from 'react';
import { FC } from 'react';
import { StyleSheet } from 'react-native';
import MapView, { Marker, Region } from 'react-native-maps';
import { useMarkerPressed } from '../../hooks/marker-pressed';
import { useMapNavigation } from '../../hooks/use-map-navigation';
import { markers } from './map-data';
import { MapMarkers } from './map-markers';

export interface IMapRegion {
  latitude: number;
  longitude: number;
  latitudeDelta: number;
  longitudeDelta: number;
}

interface IMapArea {
  region: Region;
  onRegionChangeComplete: (region: IMapRegion) => void;
}

export const MapArea: FC<IMapArea> = ({ region, onRegionChangeComplete }) => {
  const { mapRef, animateToCoordinat } = useMapNavigation();


  React.useEffect(() => {
    if (mapRef?.current) {
      mapRef?.current.fitToSuppliedMarkers(markers.map(({ id }) => id));
    }
  }, [markers]);

  return (
    <MapView
      ref={mapRef}
      style={styles.wrapper}
      region={region}
      onRegionChangeComplete={onRegionChangeComplete}
      showsUserLocation={true}
      showsPointsOfInterest={true}
      showsCompass={false}
      zoomControlEnabled={false}
    >
      <MapMarkers />
    </MapView>
  );
};


const styles = StyleSheet.create({
  wrapper: {
    alignSelf: 'stretch',
    height: '100%',
  },
});
