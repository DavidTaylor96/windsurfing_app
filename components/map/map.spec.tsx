import * as React from 'react';
import { Region } from 'react-native-maps';
import renderer from 'react-test-renderer';
import { MapArea } from './map';
import { MapMarkers } from './map-markers';

it(`Map area`, () => {
  const onRegionChange = jest.fn();

  const location: Region = {
    latitude: 55.953251,
    longitude: -3.188267,
    latitudeDelta: 5.5036,
    longitudeDelta: 5.5612,
  };

  const mapArea = renderer.create(
    <MapArea onRegionChangeComplete={onRegionChange} region={location}>
      <MapMarkers />
    </MapArea>
  );

  mapArea.root.props.onRegionChangeComplete();

  expect(mapArea.root.children.length).toBe(1);
  expect(onRegionChange).toHaveBeenCalledTimes(1);
  expect(mapArea).toMatchSnapshot();
});
