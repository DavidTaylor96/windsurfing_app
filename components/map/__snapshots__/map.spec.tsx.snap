// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map area 1`] = `
<AIRMap
  handlePanDrag={false}
  initialRegion={null}
  onChange={[Function]}
  onLayout={[Function]}
  onMapReady={[Function]}
  onMarkerPress={[Function]}
  onRegionChangeComplete={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  region={
    Object {
      "latitude": 55.953251,
      "latitudeDelta": 5.5036,
      "longitude": -3.188267,
      "longitudeDelta": 5.5612,
    }
  }
  showsCompass={false}
  showsPointsOfInterest={true}
  showsUserLocation={true}
  style={
    Object {
      "alignSelf": "stretch",
      "height": "100%",
    }
  }
  zoomControlEnabled={false}
>
  <AIRMapMarker
    coordinate={
      Object {
        "latitude": 55.953251,
        "longitude": -3.188267,
      }
    }
    identifier="123"
    onPress={[Function]}
    stopPropagation={false}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "position": "absolute",
        },
        undefined,
      ]
    }
    title="one"
  >
    <Text />
  </AIRMapMarker>
  <AIRMapMarker
    coordinate={
      Object {
        "latitude": 55.853251,
        "longitude": -2.488267,
      }
    }
    identifier="124"
    onPress={[Function]}
    stopPropagation={false}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "position": "absolute",
        },
        undefined,
      ]
    }
    title="two"
  >
    <Text />
  </AIRMapMarker>
  <AIRMapMarker
    coordinate={
      Object {
        "latitude": 55.753251,
        "longitude": -2.588267,
      }
    }
    identifier="125"
    onPress={[Function]}
    stopPropagation={false}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "position": "absolute",
        },
        undefined,
      ]
    }
    title="Three"
  >
    <Text />
  </AIRMapMarker>
</AIRMap>
`;
